PhysicsTools/PatAlgos/python/tools/jetTools.py

```diff
@@ -86,17 +90,27 @@ def setupJetCorrections(process, knownModules, jetCorrections, jetSource, pvSour
             marks) and use raw tcMET together with JPTJets.")
         ## set up jet correctors for MET corrections
         process.load( "JetMETCorrections.Configuration.JetCorrectorsAllAlgos_cff") # FIXME: This adds a lot of garbage
-
+        # I second the FIXME comment on the last line. When I counted it, this brought in 344 EDProducers
+        # to be available to run unscheduled. All jet correctors, probably some small fraction of which
+        # are actually used.
+        patAlgosToolsTask.add(process.jetCorrectorsAllAlgosTask)
         _payloadType = jetCorrections[0].split(_type)[0].lower()+_type
         if "PF" in _type :
             setattr(process, jetCorrections[0]+'L1FastJet', getattr(process, _payloadType+'L1FastjetCorrector').clone(srcRho=cms.InputTag('fixedGridRhoFastjetAll')))
+            patAlgosToolsTask.add(getattr(process, jetCorrections[0]+'L1FastJet'))
```

Comment:
Yes. I think you are right. That would be better. I will change that. It will probably be Monday before I update the PR.